osquery/tables/system/darwin/gatekeeper.cpp

```diff
@@ -0,0 +1,172 @@
+#include <boost/filesystem/path.hpp>
+
+#include <osquery/core.h>
+#include <osquery/filesystem.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+#include "osquery/core/conversions.h"
+#include "osquery/sql/sqlite_util.h"
+
+namespace fs = boost::filesystem;
+namespace pt = boost::property_tree;
+
+namespace osquery {
+namespace tables {
+
+const std::string kGkeStatusPath{
+    "/var/db/SystemPolicy-prefs.plist"};
+
+const std::string kGkeBundlePath{
+    "/var/db/gke.bundle/Contents/version.plist"};
+
+const std::string kGkeOpaquePath{
+    "/var/db/gkopaque.bundle/Contents/version.plist"};
+
+const std::string kPolicyDb = "/var/db/SystemPolicy";
+
+bool isGateKeeperDevIdEnabled() {
+  sqlite3* db = nullptr;
+  auto rc = sqlite3_open_v2(
+      kPolicyDb.c_str(),
+      &db,
+      (SQLITE_OPEN_READONLY | SQLITE_OPEN_PRIVATECACHE | SQLITE_OPEN_NOMUTEX),
+      nullptr);
+  if (rc != SQLITE_OK || db == nullptr) {
+    VLOG(1) << "Cannot open Gatekeeper DB: " << rc << " "
+            << getStringForSQLiteReturnCode(rc);
+    if (db != nullptr) {
+      free(db);
+    }
+  }
+
+  std::string query = "SELECT disabled FROM authority WHERE label = 'Developer ID'";
+  sqlite3_stmt* stmt = nullptr;
+  rc = sqlite3_prepare_v2(db, query.c_str(), -1, &stmt, nullptr);
+
+  while ((rc = sqlite3_step(stmt)) == SQLITE_ROW) {
+    int value = sqlite3_column_int(stmt, 0);
+    if (value == 1) {
+      // Clean up.
+      sqlite3_finalize(stmt);
+      free(db);
+      return false; // return false if any rows say "disabled"
+    }
+  }
+  sqlite3_finalize(stmt);
+  free(db);
+  return true;
+}
+
+QueryData genGateKeeper(QueryContext& context) {
+  Row r;
+
+  auto gke_status = SQL::selectAllFrom("plist", "path", EQUALS, kGkeStatusPath);
+
+  if (gke_status.empty()) {
+    r["assessments_enabled"] = INTEGER(0);
+  }
+
+  for (const auto& row : gke_status) {
+    if (row.at("key") == "enabled" && row.at("value") == "yes" ) {
+      r["assessments_enabled"] = INTEGER(1);
+      if (isGateKeeperDevIdEnabled()) {
+        r["dev_id_enabled"] = INTEGER(1);
+      } else {
+        r["dev_id_enabled"] = INTEGER(0);
+      }
+    } else {
+      r["assessments_enabled"] = INTEGER(0);
+      r["dev_id_enabled"] = INTEGER(0);
+    }
+  }
+
+  auto gke_bundle = SQL::selectAllFrom("plist", "path", EQUALS, kGkeBundlePath);
+
+  if (gke_bundle.empty()) {
+    r["version"] = std::string();
+  }
+
+  for (const auto& row : gke_bundle) {
+    if (row.at("key") == "CFBundleShortVersionString") {
+      r["version"] = row.at("value");
+    }
+  }
+
+  auto gke_opaque = SQL::selectAllFrom("plist", "path", EQUALS, kGkeOpaquePath);
+
+  if (gke_opaque.empty()) {
+    r["opaque_version"] = std::string();
+  }
+
+  for (const auto& row : gke_opaque) {
+    if (row.at("key") == "CFBundleShortVersionString") {
+      r["opaque_version"] = row.at("value");
+    }
+  }
+  return {r};
+}
+
+void genGateKeeperApprovedAppRow(sqlite3_stmt* stmt, Row& r) {
```

Comment:
Perhaps the intention is to make it `sqlite3_stmt* const stmt`? (constant pointer vs. pointer to constant)